src/plugins/data/public/query/query_string/query_results_service/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { PublicMethodsOf } from 'packages/osd-utility-types/target';
+import { QueryResultService } from './query_result_service';
+import { Query } from '../../../../common';
+
+export interface QueryResultExtensionConfig {
+  /**
+   * The id for the discover results extension.
```

Comment:
```suggestion
   * The id for the query results extension.
```

since the data plugin can be used by any plugin and is used in dashboards and visualizations